cmd/earth/main.go

```diff
@@ -159,6 +159,38 @@ func (app *earthApp) insertBashCompleteEntry() error {
 	return err
 }
 
+func (app *earthApp) deleteZcompdump() error {
+	var homeDir string
+	sudoUser, found := os.LookupEnv("SUDO_USER")
+	if !found {
+		currentUser, err := user.Current()
+		if err != nil {
+			return err
+		}
+		homeDir = currentUser.HomeDir
+	} else {
+		currentUser, err := user.Lookup(sudoUser)
+		if err != nil {
+			return err
+		}
+		homeDir = currentUser.HomeDir
+	}
+	files, err := ioutil.ReadDir(homeDir)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		if strings.HasPrefix(f.Name(), ".zcompdump") {
+			path := filepath.Join(homeDir, f.Name())
+			err := os.Remove(path)
+			if err != nil {
+				return err
```

Comment:
We should wrap this with `errors.Wrap` and also include the filename in the error.